server/index.js

```diff
@@ -51,6 +51,94 @@ app.get('/api/products/:productId', (req, res, next) => {
     .catch(err => next(err));
 });
 
+app.get('/api/cart', (req, res, next) => {
+  if (req.session.cartId) {
+    const sessionSql = `
+    select "c"."cartItemId",
+           "c"."price",
+           "p"."productId",
+           "p"."image",
+           "p"."name",
+           "p"."shortDescription"
+      from "cartItems" as "c"
+      join "products" as "p" using ("productId")
+     where "c"."cartId" = $1
+    `;
+    const sessionCartId = [req.session.cartId];
+    db.query(sessionSql, sessionCartId)
+      .then(data => res.json(data.rows));
+  } else {
+    const sql = `
+    select *
+      from "cartItems"
+    `;
+    db.query(sql)
+      .then(data => res.json(data.rows));
+  }
+});
+
+app.post('/api/cart/:productId', (req, res, next) => {
+  if (req.params.productId < 1) next(new ClientError('Product ID must be a positive integer', 400));
+  const sql = `
+  select "price"
+    from "products"
+   where "productId" = $1
+  `;
+  const productId = [req.params.productId];
+  db.query(sql, productId)
+    .then(data => {
+      if (!data.rows) {
+        next(new ClientError('Cannot find a product with that ID', 400));
```

Comment:
Because we're in a `Promise` chain, you should `throw` the new error object or `return Promise.reject(new ClientError(/* etc */))`

That way the error shortcuts to the `catch()` callback.